javatests/arcs/core/entity/HandleManagerTestBase.kt

```diff
@@ -217,18 +221,23 @@ open class HandleManagerTestBase {
     open fun singleton_clearOnAClearDataWrittenByB() = testRunner {
         val handleA = writeHandleManager.createSingletonHandle()
         val handleB = readHandleManager.createSingletonHandle()
-        val handleBUpdated = handleB.onUpdateDeferred()
-        handleA.store(entity1)
+        val handleBUpdated = handleB.onUpdateDeferred { it != null }
+        withContext(handleA.dispatcher) {
+            handleA.store(entity1)
+        }
```

Comment:
I think it's time to set up helpers for this. Maybe move the ones in `sdk/testing/BaseTestHarness.kt` to a util class?

Reply:
+1

Reply:
I'll look into that for a future PR

Reply:
created http://b/159612993 to track that